tools/mtouch/Assembly.cs

```diff
@@ -72,14 +72,22 @@ HashSet<string> ComputeDependencies (List<Exception> warnings)
 		}
 
 		// returns false if the assembly was not copied (because it was already up-to-date).
-		public bool CopyAssembly (string source, string target, bool copy_mdb = true)
+		public bool CopyAssembly (string source, string target, bool copy_mdb = true, bool strip = false)
```

Comment:
Not quite - that case is unlikely (should be only Xamarin.iOS.dll and because of the native types base type difference they will always differ, even after stripping).

I meant that stripping while copying makes it harder to optimize stripping of fat applications. If we strip before de-duplicating we're doing it twice for every assembly. If we de-duplicate first then we can only strip the non-symlinked assemblies (should be half + 1 in most cases). The later seems faster (but I don't have numbers to back it up).